plugins/obs-filters/nvidia-audiofx-filter.c

```diff
@@ -0,0 +1,924 @@
+			os_atomic_set_bool(&ng->reinit_done, true);
+		}
+	}
+	return true;
+
+failure:
+	ng->use_nvafx = false;
+	return false;
+}
+
+static void *nvidia_audio_initialize(void *data)
+{
+	struct nvidia_audio_data *ng = data;
+	NvAFX_Status err;
+
+	if (!ng->use_nvafx && !nvidia_afx_loaded) {
+		return NULL;
+	}
+	pthread_mutex_lock(&ng->nvafx_mutex);
+	pthread_mutex_lock(&nvidia_afx_initializer_mutex);
+	if (!nvidia_audio_initialize_internal(data)) {
+		goto failure;
+	}
+	if (ng->use_nvafx) {
+		err = NvAFX_GetU32(ng->handle[0],
+				   NVAFX_PARAM_NUM_INPUT_CHANNELS,
+				   &ng->num_channels);
+		if (err != NVAFX_STATUS_SUCCESS) {
+			do_log(LOG_ERROR,
+			       "NvAFX_GetU32() failed to get the number of channels, error %i",
+			       err);
+			goto failure;
+		}
+		if (ng->num_channels != 1) {
+			do_log(LOG_ERROR,
+			       "The number of channels is not 1 in the sdk any more ==> update code");
+			goto failure;
+		}
+		NvAFX_Status err = NvAFX_GetU32(
+			ng->handle[0], NVAFX_PARAM_NUM_INPUT_SAMPLES_PER_FRAME,
+			&ng->num_samples_per_frame);
+		if (err != NVAFX_STATUS_SUCCESS) {
+			do_log(LOG_ERROR,
+			       "NvAFX_GetU32() failed to get the number of samples per frame, error %i",
+			       err);
+			goto failure;
+		}
+		if (ng->num_samples_per_frame != NVAFX_FRAME_SIZE) {
+			do_log(LOG_ERROR,
+			       "The number of samples per frame has changed from 480 (= 10 ms) ==> update code");
+			goto failure;
+		}
+	}
+	ng->nvafx_initialized = true;
+	ng->nvafx_loading = false;
+	pthread_mutex_unlock(&nvidia_afx_initializer_mutex);
+	pthread_mutex_unlock(&ng->nvafx_mutex);
+	return NULL;
+
+failure:
+	ng->use_nvafx = false;
+	pthread_mutex_unlock(&nvidia_afx_initializer_mutex);
+	pthread_mutex_unlock(&ng->nvafx_mutex);
+	nvidia_audio_destroy(ng);
+	return NULL;
+}
+
+static inline enum speaker_layout nv_convert_speaker_layout(uint8_t channels)
+{
+	switch (channels) {
+	case 0:
+		return SPEAKERS_UNKNOWN;
+	case 1:
+		return SPEAKERS_MONO;
+	case 2:
+		return SPEAKERS_STEREO;
+	case 3:
+		return SPEAKERS_2POINT1;
+	case 4:
+		return SPEAKERS_4POINT0;
+	case 5:
+		return SPEAKERS_4POINT1;
+	case 6:
+		return SPEAKERS_5POINT1;
+	case 8:
+		return SPEAKERS_7POINT1;
+	default:
+		return SPEAKERS_UNKNOWN;
+	}
+}
+
+static void set_nv_model(void *data, const char *method)
+{
+	struct nvidia_audio_data *ng = data;
+	const char *file;
+	if (strcmp(NVAFX_EFFECT_DEREVERB, method) == 0)
+		file = NVAFX_EFFECT_DEREVERB_MODEL;
+	else if (strcmp(NVAFX_EFFECT_DEREVERB_DENOISER, method) == 0)
+		file = NVAFX_EFFECT_DEREVERB_DENOISER_MODEL;
+	else
+		file = NVAFX_EFFECT_DENOISER_MODEL;
+	size_t size = strlen(ng->sdk_path) + strlen(file) + 1;
+	char *buffer = (char *)bmalloc(size);
+
+	strcpy(buffer, ng->sdk_path);
+	strcat(buffer, file);
+	ng->model = buffer;
+}
+
+static void nvidia_audio_update(void *data, obs_data_t *s)
+{
+	struct nvidia_audio_data *ng = data;
+
+	if (!ng->use_nvafx)
+		return;
+	const char *method = obs_data_get_string(s, S_METHOD);
```

Comment:
Separate this from the `if` contents above with a blank line.

Reply:
done